### PR TITLE
Update to scala 2.12 and Java8 bytecode

### PR DESCRIPTION
**Pull request by gmokki**
Also update to sbt 0.13.15 and bump other dependencies to latest versions

### CODE REVIEW

**Comment by laurilehmijoki** [comment]
Hello Mikko! Are these changes necessary for you to keep using Havarunner?

If yes, can you please explain why.

Thanks!

**Comment by laurilehmijoki** [comment]
The Travis build fails for some reason. Care to take a look at it?

**Comment by gmokki** [comment]
The java7 build seems to crash from bug in glibc/java7 - I just pushed an update that drops the java7 build from travis.

It seems that the Java8 build fails to download the sbt 0.13.15. But trying to download the failing url manually succeeds in the download - it just takes 28 seconds before the download starts. Most likely the preinstalled sbt version in travis just does not like to wait that long.
The 0.13.15 version is given as an example sbt version to use in http://www.scala-sbt.org/0.13/docs/Travis-CI-with-sbt.html, there is some text about adding more retries and caching of $HOME/.sbt folder between travis runs. Maybe those are required with newer sbt when using travis.

**Comment by laurilehmijoki** [comment]
@gmokki What's your overall motivation for this pull request? Is something broken at your end?

**Comment by gmokki** [comment]
I started getting the following error when running tests with havarunner
`java.lang.NoSuchMethodError: scala.Predef$.refArrayOps([Ljava/lang/Object;)Lscala/collection/mutable/ArrayOps;` after I changed our project dependency to scala 2.12
Updating the havarunner to use scala 2.12 fixed them, but of course it could be just that I got some mixed transitive dependencies and fixing them up manually could be enough.

File: build.sbt [hunk 1]
```diff
@@ -8,25 +8,27 @@ crossPaths := false
...
+    Some("releases"  at nexus + "service/local/staging/deploy/maven2")
 }
 
-scalaVersion := "2.11.2"
+scalaVersion := "2.12.2"
 
 scalacOptions ++= Seq( "-deprecation", "-unchecked", "-feature", "-language:postfixOps", "-language:existentials", "-language:implicitConversions"  )
 
-javacOptions ++= Seq("-source", "1.7")
+javacOptions ++= Seq("-source", "1.8")
```
Comment (laurilehmijoki) [inline build.sbt]: Is there a problem with the `1.7` setting?

File: build.sbt [hunk 2]
```diff
@@ -8,25 +8,27 @@ crossPaths := false
 
 publishMavenStyle := true
 
-publishTo <<= version { (v: String) =>
+publishTo := {
   val nexus = "https://oss.sonatype.org/"
-  if (v.trim.endsWith("SNAPSHOT")) Some("snapshots" at nexus + "content/repositories/snapshots")
-  else Some("releases" at nexus + "service/local/staging/deploy/maven2")
+  if (isSnapshot.value)
```
Comment (laurilehmijoki) [inline build.sbt]: What is the idea behind this change?

File: build.sbt [hunk 3]
```diff
@@ -8,25 +8,27 @@ crossPaths := false
...
 
-javacOptions ++= Seq("-source", "1.7")
+javacOptions ++= Seq("-source", "1.8")
 
 parallelExecution in Test := false
 
 libraryDependencies ++= Seq(
-  "junit" % "junit" % "4.11" % "provided, test",
-  "com.google.guava" % "guava" % "14.0" % "provided",
+  "junit" % "junit" % "4.12" % "provided, test",
```
Comment (laurilehmijoki) [inline build.sbt]: Was there a problem with the `4.11` version?

File: build.sbt [hunk 4]
```diff
@@ -8,25 +8,27 @@ crossPaths := false
...
+  "com.google.guava" % "guava" % "22.0" % "provided",
```
Comment (laurilehmijoki) [inline build.sbt]: Was there a problem with the `22.0` version?

File: build.sbt [hunk 5]
```diff
@@ -8,25 +8,27 @@ crossPaths := false
...
   "com.novocode" % "junit-interface" % "0.11" % "test",
-  "com.google.code.findbugs" % "jsr305" % "2.0.1"  % "test"
+  "com.google.code.findbugs" % "jsr305" % "3.0.2"  % "test"
```
Comment (laurilehmijoki) [inline build.sbt]: Was there a problem with the `2.0.1` version?

**Comment by laurilehmijoki** [comment]
@gmokki ok thank you for the explanation. I added a few more questions about the version number changes in this pull request.

Overall, in this pull request, I would like to see the smallest set of changes that resolves your problem.

File: build.sbt (hunk 1 above)
Comment (gmokki) [inline build.sbt]: Scala 2.12 only supports Java 8, there should be no reason to compile against java 7 targe

File: build.sbt (hunk 2 above)
Comment (gmokki) [inline build.sbt]: sbt 0.13.15 no longer allowed the `<<=` syntax I replaced the code with the example from [sbt docs > sonatype > maven](http://www.scala-sbt.org/0.13/docs/Using-Sonatype.html#Maven+configuration+tips%E2%80%99n%E2%80%99tricks)

File: build.sbt (hunk 4 above)
Comment (gmokki) [inline build.sbt]: No problems, the version number won't be anyway leaking out.
The effect between old and new dependency is that do we want to verify that havarunner works with an old version (and trust that newer guava retains backwards compatibility) or do we test that it works with the latest released version.
I can revert that back if you want.
I do know that the ClassPath code is broken in guava 19, 20 and 21 versions - it fails when run inside jenkins workspaces that contain (by default) by default special characters (for example `workspace@2`)

**Comment by laurilehmijoki** [comment]
@gmokki thank you for the clarifications! I've just released this pull request in the version **1.2.2**. The package should be available at Maven central in a while.

**Pull request merged**